packages/bsky/src/services/indexing/plugins/post.ts

```diff
@@ -96,6 +103,37 @@ const insertFn = async (
     return null // Post already indexed
   }
 
+  if (obj.reply) {
+    const replyRoot = obj.reply.root.uri
+    const replyParent = obj.reply.parent.uri
+    const replyRefs = await db
+      .selectFrom('post')
+      .innerJoin('record', 'record.uri', 'post.uri')
+      .where('post.uri', 'in', [replyRoot, replyParent])
+      .selectAll('post')
+      .select('record.json')
+      .execute()
+    const root = replyRefs.find((ref) => ref.uri === replyRoot)
+    const parent = replyRefs.find((ref) => ref.uri === replyParent)
+    const isInvalidReply = !parent || checkInvalidReply(obj.reply, parent)
+    const isInvalidInteraction =
```

Comment:
may be nice to split up this logic a bit - have a hard time following it

i also think that this invalidates any replies after a root is deleted?